cli/azd/test/ostest/ostest.go

```diff
@@ -0,0 +1,89 @@
+package ostest
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/exec"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry"
+	azdexec "github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/sethvargo/go-retry"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// TempDirWithDiagnostics creates a temp directory with cleanup that also provides additional
+// diagnostic logging and retries.
+func TempDirWithDiagnostics(t *testing.T) string {
+	temp := t.TempDir()
+
+	if runtime.GOOS == "windows" {
+		// Enable our additional custom remove logic for Windows where we see locked files.
+		t.Cleanup(func() {
+			err := removeAllWithDiagnostics(t, temp)
+			if err != nil {
+				logHandles(t, temp)
+				t.Fatalf("TempDirWithDiagnostics: %s", err)
+			}
+		})
+	}
+
+	return temp
+}
+
+func logHandles(t *testing.T, path string) {
+	handle, err := exec.LookPath("handle")
+	if err != nil && errors.Is(err, exec.ErrNotFound) {
+		t.Logf("handle.exe not present. Skipping handle detection. PATH: %s", os.Getenv("PATH"))
+		return
+	}
+
+	if err != nil {
+		t.Logf("failed to find handle.exe: %s", err)
+		return
+	}
+
+	args := azdexec.NewRunArgs(handle, path, "-nobanner")
```

Comment:
I know on first run the sys internal tools (even command line tools like handle.exe) pop a dialog to make you accept an EULA.  Does the choco install also accept the eula or do we need to pass `/accepteula` to `handle` as well?

Reply:
I don't think `choco` accepts the EULA.

I did not find `accepteula` as part of the documented flags for `handle` though (in contrast to other sysinternals programs):
```
PS D:\repos\azd-sources\csharp-cosmos-sql> handle -?

Nthandle v4.22 - Handle viewer
Copyright (C) 1997-2019 Mark Russinovich
Sysinternals - www.sysinternals.com

usage: handle [[-a [-l]] [-u] | [-c <handle> [-y]] | [-s]] [-p <process>|<pid>] [name] [-nobanner]
  -a         Dump all handle information.
  -l         Just show pagefile-backed section handles.
  -c         Closes the specified handle (interpreted as a hexadecimal number).
             You must specify the process by its PID.
             WARNING: Closing handles can cause application or system instability.
  -y         Don't prompt for close handle confirmation.
  -s         Print count of each type of handle open.
  -u         Show the owning user name when searching for handles.
  -p         Dump handles belonging to process (partial name accepted).
  name       Search for handles to objects with <name> (fragment accepted).
  -nobanner  Do not display the startup banner and copyright message.
```

will leave it as-is since it's working.